[BUG] - Hide remove button from mandatory fields in editor

PascalLike opened the pull request:
In the metadata editor on some dutch schema, it happens that the remove button is shown even close to specific mandatory fields when the parent is not mandatory. For example, check https://github.com/metadata101/iso19139.nl.geografie.2.0.0

This is an extra check on the current field, to check explicitly if it's mandatory, before showing the remove button.

fxprunayre commented:
@josegar74 did you tested it ? any unexpected side effects ?

josegar74 commented:
I haven't tested yet, will check and update with the results.

fxprunayre commented:
Also, why not simply decide to do it on a per view basis (custom & simple) ? and maybe simply using CSS `.gn-required > div.col-sm-1` hidden. So you keep the possibility to remove 0:N elements in the advanced view at least.

PascalLike commented on web/src/main/webapp/xslt/ui-metadata/form-builder.xsl:
```diff
@@ -1299,15 +1301,17 @@
+                $editInfo/@del = 'true' or
+                $editInfo/@min != 1
+              )
+            )) and not($isRequired and ($editInfo and $editInfo/@max and $editInfo/@max = 1) and ($parentEditInfo and $parentEditInfo/@max and $parentEditInfo/@max = 1))">
```
The conditions ($editInfo and $editInfo/@max and $editInfo/@max = 1) and ($parentEditInfo and $parentEditInfo/@max and $parentEditInfo/@max = 1) are between parentesis for readability, because it's not possible to check directly $elem/@max = 1 without getting xpath errors.

PascalLike commented:
I added an extra condition to avoid the remove on 0:N, like found on gmd:presentationForm (to test I added the condition mandatory to this field)

![mandatorywith](https://user-images.githubusercontent.com/1323093/53172559-e8bd1700-35e5-11e9-98e6-dc9cc4ee0196.png)

Still working for 0:1, for example on "Algemene beschrijving herkomst" (visible only in dutch on standard view iso19139.nl.geografie.2.0.0)

![mandatorywithout](https://user-images.githubusercontent.com/1323093/53172695-42bddc80-35e6-11e9-8d88-e5eeabe9c9b3.png)

Delawen commented:
The code looks good and I don't see any obvious bug. But I couldn't find a field to test this on basic ISO19139 to compare the behaviour. Is there any field that before this fix showed the delete button and not anymore now?

Delawen commented:
I tried making the gmd:alternateTitle mandatory:

>   <element name="gmd:alternateTitle" id="361.0">
>     <label>Alternate title</label>
>     <condition>mandatory</condition>
>     <description>Short name or other language name by which the cited information is known.
>       Example: "DCW" as an alternative title for "Digital Chart of the World
>     </description>
>   </element>

but I'm still getting a remove button.That's because it is multiple and is out of scope of this, right?

PascalLike commented:
Right, this case falls in the scope of a client side check. The fix should be here: https://github.com/geonetwork/core-geonetwork/pull/3013

Delawen commented:
Tried with gmd:purpose and works perfectly! Thanks.
